From: James Smart <jsmart2021@gmail.com>
Date: Sat, 13 May 2017 12:07:15 -0700
Subject: nvme core: allow controller RESETTING to RECONNECTING transition
References: bsc#1037838
Patch-Mainline: submitted to linux-scsi 2017/05/16

Allow controller state transition : RESETTING to RECONNECTING

The nvme_fc transport will set the state to RESETTING when
tearing down the current association (error or controller reset),
then transition to RECONNECTING when attempting to establish a new
association.

Signed-off-by: James Smart <james.smart@broadcom.com>
Reviewed-by: Hannes Reinecke <hare@suse.com>
---
 drivers/nvme/host/core.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/nvme/host/core.c b/drivers/nvme/host/core.c
index 17fb694..0c75ebd 100644
--- a/drivers/nvme/host/core.c
+++ b/drivers/nvme/host/core.c
@@ -163,6 +163,7 @@ bool nvme_change_ctrl_state(struct nvme_ctrl *ctrl,
 	case NVME_CTRL_RECONNECTING:
 		switch (old_state) {
 		case NVME_CTRL_LIVE:
+		case NVME_CTRL_RESETTING:
 			changed = true;
 			/* FALLTHRU */
 		default:
-- 
1.8.5.6

